nixos/modules/services/web-apps/librechat.nix

```
@@ -0,0 +1,157 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.librechat;
+  format = pkgs.formats.yaml { };
+  configFile = format.generate "librechat.yaml" cfg.settings;
+  # Thanks to https://github.com/nix-community/home-manager/blob/60e4624302d956fe94d3f7d96a560d14d70591b9/modules/lib/shell.nix :)
+  export = n: v: ''export ${n}="${builtins.toString v}"'';
+  exportAll = vars: lib.concatStringsSep "\n" (lib.mapAttrsToList export vars);
+  exportCredentials = n: _: ''export ${n}="$(${pkgs.systemd}/bin/systemd-creds cat ${n}_FILE)"'';
+  exportAllCredentials = vars: lib.concatStringsSep "\n" (lib.mapAttrsToList exportCredentials vars);
+  transformCredential = n: v: "${n}_FILE:${v}";
+  getLoadCredentialList = lib.mapAttrsToList transformCredential cfg.credentials;
+in
+{
+  options.services.librechat = {
+    enable = lib.mkEnableOption "Whether to enable the LibreChat server.";
+    package = lib.mkPackageOption pkgs "librechat" { };
+    openFirewall = lib.mkEnableOption "Whether to open the port in the firewall.";
```

If a description would sound odd when prefixed with "Whether to enable...", you could also do this

```suggestion
    openFirewall = lib.mkEnableOption null // {
      description = "Whether to open the port in the firewall.";
    };
```